src/components/utils/ManagedTree.js

```diff
@@ -31,6 +50,13 @@ let ManagedTree = React.createClass({
     }
   },
 
+  expandListItems(itemsList) {
+    const key = this.props.getKey(itemsList[itemsList.length - 1]);
+    const expanded = this.state.expanded;
+    expanded.add(key);
+    this.setState({ expanded: expanded });
```

Comment:
same weirdness here. maybe we can do something where we look at the listItem props and then update the expanded set in batch?